model/host/host.go

```diff
@@ -1194,7 +1222,18 @@ func StaleRunningTaskIDs(staleness time.Duration) ([]task.Task, error) {
 		}},
 		{"$match": bson.M{
 			"$expr": bson.M{
-				"$eq": []string{"$_id", bsonutil.GetDottedKeyName("$host", RunningTaskKey)},
+				"$and": []bson.M{
+					{"$eq": []string{"$_id", bsonutil.GetDottedKeyName("$host", RunningTaskKey)}},
+					{"$or": []bson.M{ // this expression checks that the host is not currently running teardown_group of a different task
+						{"$not": []bson.M{{"$ifNull": []interface{}{bsonutil.GetDottedKeyName("$host", RunningTeardownForTaskKey), nil}}}},
+						{"$eq": []string{bsonutil.GetDottedKeyName("$host", RunningTeardownForTaskKey), ""}},
+						{"$and": []bson.M{
+							{"$ifNull": []interface{}{bsonutil.GetDottedKeyName("$host", RunningTeardownForTaskKey), nil}},
```

Comment:
I read this as: both the RunningTeardownForTaskKey is empty and the time has expired. Won't it always be true that the time has expired if the task key is empty?

Reply:
The first line is actually the opposite - it will be truthy if the task ID is set (again me trying to do a $exists). So this statement says "if the host is running teardown_group for a task and it has been over 35 minutes since it started running"

Reply:
Ah, got it, thanks.